api/open_saves.proto

```diff
@@ -430,6 +441,16 @@ message SortOrder {
   string user_property_name = 4;
 }
 
+message GetMultiRecordEntity {
+  Record record = 1;
+  string store_key = 2;
+  int32 index = 3;
```

Comment:
Could you explain why the index is necessary here?

Reply:
The index is returned to tell the caller which entry in the requested array this corresponds to.

Alternatively, I can also return the error for each requested key, but I'll have to introduce a different message to return either the error message or the record.

Which one would you prefer? Do you have any other idea of how to handle these errors?

Reply:
I see. Yeah I thought we were returning the `datastore.MultiError` at the same time, but I realized that wasn't happening. I think the best way to do this is by returning a grpc status with ErrDetails, populating the metadata with each key that returned an error. Then on the client side, you can unwrap this error and determine which keys failed.

Reply:
I like the idea of having a errorDetails array with keys and error passed back to the client in the response object.

I'll work on that to have something to review shortly.

Reply:
I think the only annoying thing with that approach is you would have to manually check if all keys failed to return an error (coding the last bullet point). I slightly prefer `ErrorInfo` but I'd be ok with either solution.

Reply:
I like the Status approach. I would argue that regardless of the number of errors, we should be returning a status per record key since the error might vary and be specific per key.
I have looked at returning an array of errorDetails, but for it to be useful it needs to include store and record keys to match the error to the requested records.

Reply:
@yuryu By Status, what data type are you implicitly referring to?

Reply:
Something like

```proto
import "google/grpc/status.proto";

// ...
  rpc GetRecord(GetRecordRequest) returns (Record) {
    option deprecated = true;
  }
  rpc GetRecords(GetRecordsRequest) returns (GetRecordsResponse) {}
// ...

message GetRecordsRequest {
  repeated string store_keys = 1;
  repeated string keys = 2;
  Hint hint = 3;
}

message GetRecordsResponse {
  message Result {
    google.rpc.Status status = 1;
    string store_key = 2;
    Record record = 3;
  }
  repeated Result results = 1;
}
```

Reply:
@yuryu I am having a hard time figuring out where to pull the status.proto file from. It seems to be included as part of rpc and also as part of googleapis. Would it be better to redefine in this project either as another file in ./api or in-line in the open_saves.proto definition?

Reply:
I ended up defining locally in the api folder as it was defined in the googleapis (without having to reference / include the whole module in open-saves).